### DISCUSSION TITLE
WiX v4 bootstrapper - setting the InstallFolder to program files doesn't resolve

### DISCUSSION THREAD

**Discussion by george7378** [wixtoolset/issues#8642] [Questions] [answered]
I have a WiX v4 project which uses a bootstrapper, declared in a `<Bundle>`. I want to set the default InstallFolder based on the program files folder, so I've structured it like this:

```
<Wix ...>
    <Bundle ...>

        <Variable Name="InstallFolder" Type="string" Value="[ProgramFilesFolder]MyProduct" />
        
        <BootstrapperApplication>
            <bal:WixStandardBootstrapperApplication Theme="hyperlinkLicense" LicenseUrl="" />
        </BootstrapperApplication>

        <Chain>
            ...
        </Chain>

    </Bundle>
</Wix>
```

However the `[ProgramFilesFolder]` doesn't resolve, and all I see is this:

![image](https://github.com/user-attachments/assets/dade48f1-97de-492d-9a9b-e84ccf36e192)

I've seen a couple of examples online suggesting this is how it's done. What am I doing wrong?

**Comment by bevanweiss** [comment]
You might want to have a dig around in the source of the WixStandardBootstrapperApplication.
I'm not sure if it's actually formatting the string it gets from InstallFolder prior to populating the Editbox

https://github.com/wixtoolset/wix/blob/main/src/ext/Bal/stdbas/WixStandardBootstrapperApplication.cpp
https://github.com/wixtoolset/wix/blob/main/src/ext/Bal/stdbas/Resources/HyperlinkTheme.xml

If doing it yourself, the guidance would be in line with this Stackoverflow post
https://stackoverflow.com/questions/17901477/how-to-get-programfilesfolder-to-populate-to-c-program-files-burn-ui-projec

**Reply by george7378** [reply]
Thanks for the reply - looking at the WixStandardBootstrapperApplication link you posted, it seems there's a constant at the top called `WIXSTDBA_VARIABLE_INSTALL_FOLDER` representing the `InstallFolder` variable which isn't used anywhere.

Looking at the [same file in V3](https://github.com/wixtoolset/wix3/blob/master/src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp#L2518), that string _does_ appear to be formatted. I wonder if this was intentionally taken out for some reason?

**Reply by bevanweiss** [reply]
I'm not even seeing the Options page listed in the current WixStandardBA handling...
https://github.com/wixtoolset/wix/blob/ce73352b1fa1d4f9cded10a0ee410f2e786bd326/src/ext/Bal/stdbas/WixStandardBootstrapperApplication.cpp#L72-L81

Delta from StdBA -> Theme is as below.  The 'missing' items in the Theme I think aren't an issue at all.  But the reverse I could see being a little tricky around things requiring such value modification (i.e. expanding formatted variables)
```diff
  "Loading"
  "Help"
  "Install"
-  "Modify"
+  "Options"
  "Progress"
-  "ProgressPassive"
  "Success"
  "Failure"
```

This might have been an oversight on the migration from v3 -> v4+, or a deliberate choice.
@barnson could probably advise better.

Since it's purely about value formatting, it probably would be nicer to push it into the 'theme' itself.. but it looks like there's not hooks to do that currently in place.
https://wixtoolset.org/docs/schema/thmutil/
Almost like each item that accepts 'variables' would also have a boolean flag that would be 'ExpandFormatted="Yes/[No]"' or such to handle these situations..

**Comment by MartQC** [comment, marked as answer]
Change your variable Type="string" to Type="formatted" and the variable [ProgramFilesFolder] will be resolved

**Reply by bevanweiss** [reply]
https://wixtoolset.org/docs/schema/wxs/variable/

reference from schema

**Reply by george7378** [reply]
Yes, just tried it and it works - thanks!
